### PR TITLE
Don't use the RC versions of sqlsrv/pdo_sqlsrv in AppVeyor

### PR DESCRIPTION
**Pull request by mlocati**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | 

#### Summary

AppVeyor is currently configured to test the 5.2.0rc1 versions of the `sqlsrv`/`pdo_sqlsrv` PHP extensions.
BTW the very latest version is now [5.2.0](https://github.com/Microsoft/msphpsql/releases): let's use it.
This is done by using the [PhpManager](https://github.com/mlocati/powershell-phpmanager) PowerShell module, that's able to automatically install the most recent stable version of PHP extensions, without the need of manually specifying their download URL and version.

### CODE REVIEW

**Comment by mlocati** [comment]
To speed up AppVeyor we could also install xdebug only if `$Env:coverage` is `yes` (otherwise it's not needed).

File: .appveyor.yml [hunk 1]
```diff
@@ -101,20 +96,12 @@ install:
...
           }
-
-          # download CA bundle
-          if (!(Test-Path C:\tools\cacert)) {
-            New-Item -path c:\tools\ -name cacert -itemtype directory
-          }
-          if (!(Test-Path c:\tools\cacert\bundle.pem)) {
-            appveyor-retry appveyor DownloadFile https://curl.haxx.se/ca/cacert.pem -Filename C:\tools\cacert\bundle.pem
-          }
         }
```
Comment (Majkl578) [inline .appveyor.yml]: Just noticed now, PHP block should end before installing Composer/Ocular - could you please move it above `download Composer`? Thanks.

**Comment by Majkl578** [comment]
> AppVeyor is currently configured to test the 5.2.0rc1

That is because we started testing against PHP 7.2 at the time when no stable PECL version was compatible.

Otherwise I can't review the rest since I am not a Windows guy.

File: .appveyor.yml (hunk 1 above)
Comment (mlocati) [reply .appveyor.yml]: What about `appveyor-retry cinst -y sqlite`?

**Comment by mlocati** [comment]
>> AppVeyor is currently configured to test the 5.2.0rc1
> That is because we started testing against PHP 7.2 at the time when no stable PECL version was compatible.

Yes, the 5.2.0 (not RC) is quite recent.

File: .appveyor.yml (hunk 1 above)
Comment (Majkl578) [reply .appveyor.yml]: That should stay as part of the PHP block.
The question is whether we even need it, we don't test against SQLite (although we originally wanted to). Imho we can drop SQLite completely.

**Comment by morozov** [comment]
The pull request header looks misleading in the way that it's not about the RC versions but is about introducing PhpManager. BTW, the extension versions are updated in https://github.com/doctrine/dbal/pull/3151 which looks like a much simpler change.

To the question of using PhpManager, given that our installation process is scripted anyways, what's the real use of having another CI dependency? Instead of just using the PHPs own way of configuring extensions, we'll use a 3rd party tool.

**Comment by mlocati** [comment]
The only reason to use PhpManager is that's really easy to use. BTW obviously it's not mandatory.
Should I close this PR?

**Comment by mlocati** [comment]
That's why there are libraries maintained by people that knows a specific language better...

**Comment by morozov** [comment]
From what I can see, it uses the same PowerShell language but instead of operating on the level of PECL packages and php.ini configuration which everyone knows about it has its own API. This API should be convenient for a daily use as a developer or for scripting different environments but in our case, we have just one environment which is static.

**Comment by mlocati** [comment]
Yes, they are both PowerShell commands, but PhpManager abstracts pear/pecl, and make it very easy to write and maintain PHP installations.

For example, here's how the current DBAL script is translated to PhpManager:

- From  
```powershell
appveyor-retry cinst --params '""/InstallDir:C:\tools\php""' --ignore-checksums -y php --version ((choco search php --exact --all-versions -r | select-string -pattern $env:php | sort { [version]($_ -split '\|' | select -last 1) } -Descending | Select-Object -first 1) -replace '[php|]','')
copy php.ini-production php.ini
Add-Content php.ini "`n date.timezone=UTC"
Add-Content php.ini "`n extension_dir=ext"
```
- To  
```powershell
Install-Php -Version $env:php -Architecture $env:platform -ThreadSafe $false -Path c:\tools\php -TimeZone UTC -AddToPath System -InitialPhpIni Production
```
- From
```powershell
if (!(Test-Path C:\tools\cacert)) {
  New-Item -path c:\tools\ -name cacert -itemtype directory
}
if (!(Test-Path c:\tools\cacert\bundle.pem)) {
  appveyor-retry appveyor DownloadFile https://curl.haxx.se/ca/cacert.pem -Filename C:\tools\cacert\bundle.pem
}
Add-Content php.ini "`n curl.cainfo=C:\tools\cacert\bundle.pem"
```
- To  
```powershell
Update-PhpCAInfo
```
- From
```powershell
Add-Content php.ini "`n memory_limit=1G"
```
- To  
```powershell
Set-PhpIniKey -Key memory_limit -Value 1G
```
- From
```powershell
Add-Content php.ini "`n extension=php_openssl.dll"
Add-Content php.ini "`n extension=php_mbstring.dll"
Add-Content php.ini "`n extension=php_fileinfo.dll"
Add-Content php.ini "`n extension=php_pdo_sqlite.dll"
Add-Content php.ini "`n extension=php_sqlite3.dll"
Add-Content php.ini "`n extension=php_curl.dll"
```
- To  
```powershell
Enable-PhpExtension -Extension openssl
Enable-PhpExtension -Extension mbstring
Enable-PhpExtension -Extension fileinfo
Enable-PhpExtension -Extension pdo_sqlite
Enable-PhpExtension -Extension sqlite3
Enable-PhpExtension -Extension curl
```
- From
```powershell
$DLLVersion = "5.2.0rc1"
cd c:\tools\php\ext
$source = "https://windows.php.net/downloads/pecl/releases/sqlsrv/$($DLLVersion)/php_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc15-x64.zip"
$destination = "c:\tools\php\ext\php_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc15-x64.zip"
Invoke-WebRequest $source -OutFile $destination
7z x -y php_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc15-x64.zip > $null
$source = "https://windows.php.net/downloads/pecl/releases/pdo_sqlsrv/$($DLLVersion)/php_pdo_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc15-x64.zip"
$destination = "c:\tools\php\ext\php_pdo_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc15-x64.zip"
Invoke-WebRequest $source -OutFile $destination
7z x -y php_pdo_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc15-x64.zip > $null
$DLLVersion = "2.6.0"
$source = "https://xdebug.org/files/php_xdebug-$($DLLVersion)-$($env:php)-vc15-nts-x86_64.dll"
$destination = "c:\tools\php\ext\php_xdebug.dll"
Invoke-WebRequest $source -OutFile $destination
Remove-Item c:\tools\php\* -include .zip
Add-Content php.ini "`nextension=php_sqlsrv.dll"
Add-Content php.ini "`nextension=php_pdo_sqlsrv.dll"
Add-Content php.ini "`nzend_extension=php_xdebug.dll"
```
- To  
```powershell
Install-PhpExtension -Extension sqlsrv
Install-PhpExtension -Extension pdo_sqlsrv
Install-PhpExtension -Extension xdebug
```

Furthermore, disabling a PHP extension using `Get-Content`/`Set-Content` would be quite complex, whereas with PhpManager you only have to call `Disable-PhpExtension`.

About the problem of learning a new tool, well, it's not that hard with PowerShell (the first commands you learn when developing PS scripts are `Get-Command -Module <module>` to list the commands offered by a module, and `Get-Help <command>` to get extensive help about that command).

TL;DR: I thought that using PhpManager would have made maintainers' life easier, but it seems it's not the case: so please feel absolutely free to close this PR :wink:

**Comment by morozov** [comment]
Sorry, I'm going to close it.
